src/main/java/it/gov/pagopa/pu/registry/mapper/debtposition/DebtPositionRegistryMapperService.java

```diff
@@ -0,0 +1,57 @@
+package it.gov.pagopa.pu.registry.mapper.debtposition;
+
+import it.gov.pagopa.pu.registry.event.payments.dto.*;
+import it.gov.pagopa.pu.registry.model.DebtPositionRegistry;
+import it.gov.pagopa.pu.workflowhub.dto.generated.DebtPositionDTO;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class DebtPositionRegistryMapperService {
+
+  private final DebtPositionEventDTO2DebtPositionRegistryMapper debtPositionEventDTO2DebtPositionRegistryMapper;
+  private final DebtPositionIoEventDTO2DebtPositionRegistryMapper debtPositionIoEventDTO2DebtPositionRegistryMapper;
+  private final DebtPositionSendEventDTO2DebtPositionRegistryMapper debtPositionSendEventDTO2DebtPositionRegistryMapper;
+
+  public DebtPositionRegistry map(PaymentEventDTO<?> event) {
+    if (event == null) {
+      log.warn("Received null event");
+      return null;
+    }
+
+    if (event.getPayload() == null) {
+      log.warn("Received event with null payload: eventType={} eventId={} traceId={} eventDescription={}",
+               event.getEventType(), event.getEventId(), event.getTraceId(), event.getEventDescription());
+      return null;
+    }
+
+    if (event.getEventType() == null) {
+      log.warn("Received event with null event type: eventId={} traceId={} eventDescription={}",
+               event.getEventId(), event.getTraceId(), event.getEventDescription());
+      return null;
+    }
+
+    Object payload = event.getPayload();
+
+    switch (payload) {
+      case DebtPositionDTO debtPositionDTO -> {
+        return debtPositionEventDTO2DebtPositionRegistryMapper.map((DebtPositionEventDTO) event);
+      }
+      case DebtPositionIoNotificationDTO debtPositionIoNotificationDTO -> {
+        return debtPositionIoEventDTO2DebtPositionRegistryMapper.map((DebtPositionIoEventDTO) event);
+      }
+      case DebtPositionSendNotificationDTO debtPositionSendNotificationDTO -> {
+        return debtPositionSendEventDTO2DebtPositionRegistryMapper.map((DebtPositionSendEventDTO) event);
+      }
+      default -> {
+        log.warn("Unsupported payload type: class={} eventType={} eventId={} traceId={} eventDescription={}", payload.getClass().getName(),
+          event.getEventType(), event.getEventId(), event.getTraceId(), event.getEventDescription());
+        return null;
+      }
```

Comment:
let's switch directly on the event type instead of casting
```suggestion
    switch (event) {
      case DebtPositionEventDTO debtPositionEvent -> {
        return debtPositionEventDTO2DebtPositionRegistryMapper.map(debtPositionEvent);
      }
      case DebtPositionIoEventDTO ioEvent -> {
        return debtPositionIoEventDTO2DebtPositionRegistryMapper.map(ioEvent);
      }
      case DebtPositionSendEventDTO sendEvent -> {
        return debtPositionSendEventDTO2DebtPositionRegistryMapper.map(sendEvent);
      }
      default -> {
        log.error("Unsupported payload type: class={} eventType={} eventId={} traceId={} eventDescription={}", payload.getClass().getName(),
          event.getEventType(), event.getEventId(), event.getTraceId(), event.getEventDescription());
        return null;
      }
```

Reply:
@antonioT90 

there are _n_ event types for a single class and we already have the sub-types defined on the attribute itself, so I do not believe it is good to refactor in that way...my proposals are to either:

1. leave it as is (it's not a casting but a pattern matching introduced in java 17, the same as **instanceof** is just an operator, so it has little to no impact on performance and it's a clean set of code)
2. create a constant with the definitions for each type of event payload and use that both in the _names_ parameter for the sub-types and then in this method

Again, I would go with number 1...so let me know.